src/main/java/Parser.java

```diff
@@ -0,0 +1,93 @@
+import command.*;
+import exceptions.DukeException;
+import task.Deadline;
+import task.Event;
+import task.TaskList;
+import task.Todo;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    Duke duke;
+
+    public Parser(Duke duke) {
+        this.duke = duke;
+    }
+
+    /**
+     * Parses the user input and returns a Command to be executed.
+     *
+     * @param input String input from user.
+     * @param tasks TaskList being used to handle tasks.
+     * @return Command to be executed.
+     * @throws DukeException if user input is invalid or improperly formatted.
+     */
+    public Command parse(String input, TaskList tasks) throws DukeException {
```

Comment:
You could consider changing parse to parseInput instead. So it is clearer that this method is used to parse the user input.